### DISCUSSION TITLE
Auto Imports for Composables in the Server Directory

### DISCUSSION THREAD

**Discussion by JordashTalon** [nuxt/nuxt#16910] [Questions]
I was looking through the docs and notices that Composables don't auto import into the server directory, I was wondering if the plan is that when the release is live it is planned to auto import composables into the server directory? Trying to plan out our structure for this.

![image](https://user-images.githubusercontent.com/957100/188242903-aa9399d0-f93d-440a-bc37-d114d871d4e6.png)

**Comment by danielroe** [comment]
Your vue composables will not work in your server routes, even if they were to be auto-imported. (Vue composables often rely on Nuxt/Vue context which won't be present in the server directory.) So what we're likely to do is a _separate_ auto-imported folder within `~/server`, such as `~/server/utils` or equivalent.

**Reply by JordashTalon** [reply]
Ok awesome that would be great

**Reply by JordashTalon** [reply]
I was looking through our current use of composables and plugins for a project we’re launching in Nuxt3 and we do have quite a few items that are used exclusively in front end / or exclusively used in the backend.

However there is quite a few that are used for both frontend and backend. Things like string manipulation functions, date manipulation, or endpoint request transformations.

I’m wondering if it would make sense to make some kind of shared folder where frontend and backend can share composables / plugins.  Or make them shared by default and if you don’t end up using it for an endpoint or visa versa it’s a non issue, as long as it doesn’t impact performance.

Another option that would work, is I was wondering if there was some way to auto / globally import commonly used modules or composables for use in any endpoint, that would also be a solve and be extremely useful.

See this question:
https://stackoverflow.com/questions/73628691/is-it-possible-to-auto-import-file-into-every-server-endpoint-for-nuxt3
